### ISSUE TITLE
`break` in `while` body always considered active even if condition is (possibly) false

### ISSUE THREAD

**Issue by samueltardieu** [rust-lang-nursery/rust-clippy#1991]
The following code

``` rust
pub fn h() {
    'label: loop {
        while false {
            break 'label;
        }
    }
}
```

triggers a false positive for the outer loop:

```
warning: this loop never actually loops
 --> src/lib.rs:2:5
  |
2 | /     'label: loop {
3 | |         while false {
4 | |             break 'label;
5 | |         }
6 | |     }
  | |_____^
```

It gives the right diagnostic for the inner `while` loop though (which indeed does not loop), but the one for the outer loop is definitely wrong as this is an infinite loop.

**Comment by oli-obk** [comment]
We need to stop iterating into inner loops known to not loop. Won't catch `while  ultracomplexfunction()  {}`, but should be good enough
